pkg/kallsyms/kallsyms_test.go

```diff
@@ -0,0 +1,177 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kallsyms
+
+import (
+	"errors"
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/btf"
+
+	utilstest "github.com/inspektor-gadget/inspektor-gadget/internal/test"
+)
+
+func TestCustomKAllSyms(t *testing.T) {
+	kAllSymsStr := strings.Join([]string{
+		"0000000000000000 A fixed_percpu_data",
```

Comment:
nit: test when `unknown` is return. (see my other comment.)

Reply:
But since there is an entry with address 0, it will never return unknown in practice.